pkg/olm/installplan.go

```diff
@@ -0,0 +1,162 @@
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the installplan is empty")
+
+		builder.errorMsg = "installplan 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the installplan is empty")
+
+		builder.errorMsg = "installplan 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Create makes an InstallPlanBuilder in cluster and stores the created object in struct.
+func (builder *InstallPlanBuilder) Create() (*InstallPlanBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the InstallPlan %v in namespace %v",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.InstallPlans(builder.Definition.Namespace).Create(context.Background(),
+			builder.Definition, metaV1.CreateOptions{})
+	}
+
+	return builder, err
+}
+
+// Exists checks whether the given installplan exists.
+func (builder *InstallPlanBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if installplan %s exists", builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.apiClient.InstallPlans(builder.Definition.Namespace).Get(
+		context.Background(), builder.Definition.Name, metaV1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Delete removes a installplan.
+func (builder *InstallPlanBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting installplan %s in namespace %s", builder.Definition.Name,
+		builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil
+	}
+
+	err := builder.apiClient.InstallPlans(builder.Definition.Namespace).Delete(context.TODO(),
+		builder.Object.Name, metaV1.DeleteOptions{})
+
+	if err != nil {
+		return err
+	}
+
+	builder.Object = nil
+
+	return err
+}
+
+// Update modifies the existing InstallPlanBuilder with the InstallPlan definition in InstallPlanBuilder.
+func (builder *InstallPlanBuilder) Update() (*InstallPlanBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating installPlan %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.apiClient.InstallPlans(builder.Definition.Namespace).Update(
+		context.TODO(), builder.Definition, metaV1.UpdateOptions{})
+
+	return builder, err
+}
+
+// validate will check that the builder and builder definition are properly initialized before
+// accessing any member fields.
+func (builder *InstallPlanBuilder) validate() (bool, error) {
+	resourceCRD := "installplan"
+
+	if builder == nil {
+		glog.V(100).Infof("The builder %v is uninitialized", resourceCRD)
```

Comment:
"The builder %s is uninitialized"

Reply:
for validate() method it's %v always was used. Are you sure I need to change?

Reply:
Not sure. Just less indirection.
resourceCRD is a string  -just clearer.